framework/src/media/MediaWorker.cpp

```diff
@@ -59,9 +59,43 @@ void MediaWorker::stopWorker()
 	}
 }
 
-MediaQueue& MediaWorker::getQueue()
+std::function<void()> MediaWorker::deQueue()
 {
-	return mWorkerQueue;
+	return mWorkerQueue.deQueue();
+}
+
+bool MediaWorker::mediaProcessHook()
+{
+	return false;
+}
+
+void MediaWorker::mediaProcess()
+{
+}
+
+void MediaWorker::commandProcess()
+{
+	std::function<void()> run = deQueue();
+	medvdbg("MediaWorker : deQueue\n");
+	if (run != nullptr) {
+		run();
+	}
+}
+
+int MediaWorker::mediaLooper()
+{
+	medvdbg("MediaWorker : mediaLooper\n");
+
+	while (mIsRunning) {
+		if (mediaProcessHook()) {
+			do {
```

Comment:
I'm not sure whether to add "do~while" is correct or not.

In the old logic, a node in "mWorkerQueue" is processed only when "mCurPlayer && (mCurPlayer->mCurState == PLAYER_STATE_PLAYING".

But in this logic, once the status of the player becomes "PLAYER_STATE_PLAYING", all nodes are processed even the status of the player is changed... it it true?

Reply:
State is changed only by command.
If any command is inserted, mWorkerqueue is not empty. So the loop will be broken.